app/src/main/java/com/gpayinterns/chat/LoginActivity.java

```diff
@@ -207,11 +198,88 @@ private void setCurrentUser()
         mEditor.putString("currentUser", currentUser).apply();
     }
 
-    private boolean authenticate(String email, String password)
+
+    private void authenticateFromServer() throws JSONException
     {
-        //TODO authenticate from backend server
-        //currentUser = userID received from the server
-        return true;
+        String userName = usernameEditText.getText().toString();
+        String password = passwordEditText.getText().toString();
+
+        String URL = "https://gcp-chat-service.an.r.appspot.com/login";
```

Comment:
Shouldn't this be a class level constant?
or rather, the hostname part should be a constant in a separate file which will be shared among different components like login, chat, etc.
We shouldn't hardcode it everywhere.

Reply:
Ok, I’ll create a separate public class, with which I can access all such details

Reply:
Ah! the changes involving this have been done on a separate branch. It's PR is named as: "Android: Encapsulated Server URL Details + view time for messages"